### ISSUE TITLE
HappyThread: expected loader to be pending on source file

### ISSUE THREAD

**Issue by JamieMason** [amireh/happypack#60]
Hi @amireh,
Thanks for this great tool. 

I was wondering if you could help me please, I'm faced with an `AssertionError: HappyThread: expected loader to be pending on source file` but don't understand from this message what is going on or what I can do to resolve it. I'm probably missing something obvious.

As a guess, I thought this might be some kind of locking or race condition, so I tried setting `threads` to `1` – but that had no effect.

```js
{
  AssertionError: HappyThread: expected loader to be pending on source file '/Users/jmason/Development/project/src/app/Routes/InPlay.js' (this is likely an internal error!)
  at ChildProcess.acceptMessageFromWorker(/Users/jmason/Development/project/node_modules/happypack/lib/HappyThread.js:37:11)
  at emitTwo(events.js:111:20)
  at ChildProcess.emit(events.js:191:7)
  at process.nextTick(internal/child_process.js:719:12)
  at _combinedTickCallback(internal/process/next_tick.js:67:7)
  at Immediate._tickCallback[as _onImmediate](internal/process/next_tick.js:98:9)
  at tryOnImmediate(timers.js:543:15)
  at processImmediate[as _immediateCallback](timers.js:523:5)
  name: 'AssertionError',
  actual: false,
  expected: true,
  operator: '==',
  message: 'HappyThread: expected loader to be pending on source file \'/Users/jmason/Development/project/src/app/Routes/InPlay.js\' (this is likely an internal error!)',
  generatedMessage: false
}
```

Appreciate your time, thank you.

**Comment by amireh** [comment]
Hi! So what the error actually indicates is that we've compiled a file (in the background) but the foreground loader is no longer expecting any files. It might be a sign that a file got compiled twice, or some really weird internal error.

Can you provide more context like webpack configs, source files, etc.? Thanks!

**Comment by amireh** [comment]
What kind of loaders is `Routes/IsPlay.js` being passed through? Can you show the chain?

**Comment by JamieMason** [comment]
Sorry for leaving you hanging, I'd been trying various things while still
at work earlier and I've not been sure which parts would be useful to show
and which would not. The project is private too which complicates things
further.

That file is babel and react but the file it complains about tends to vary
each run.

I'll pick it back up again on Monday. Thanks for trying to help, sorry for
the lack of information.

**Comment by derrikcurran** [comment]
I had the same issue when I had both preLoaders and loaders sharing the same thread pool.

**Comment by amireh** [comment]
This should be fixed in the 2.2.1 release, please confirm and close the ticket if so.

**Comment by amireh** [comment]
Closing this as it should've been fixed, but would still be nice to receive a confirmation.
